Sources/LanguageServerProtocolJSONRPC/MessageCoding.swift

```diff
@@ -77,7 +77,15 @@ extension JSONRPCMessage: Codable {
           throw MessageDecodingError.methodNotFound(method)
         }
 
-        let params = try messageType.init(from: container.superDecoder(forKey: .params))
+        let params: _RequestType
+        do {
+          params = try messageType.init(from: container.superDecoder(forKey: .params))
+        } catch DecodingError.keyNotFound(CodingKeys.params, _) where messageType == ShutdownRequest.self {
```

Comment:
This is kind of a layering violation to have knowledge of specific requests, but I think it's fine for a workaround.

Reply:
Filed rdar://92254952 to track removing the workaround when SR-16097 is resolved.